provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/test.yml

```
@@ -39,29 +39,50 @@ jobs:
         #{{- end }}#
         persist-credentials: false
     - name: Checkout p/examples
+      #{{- if not .Config.Shards }}#
       if: matrix.testTarget == 'pulumiExamples'
+      #{{- end }}#
       uses: #{{ .Config.ActionVersions.Checkout }}#
       with:
         repository: pulumi/examples
         path: p-examples
     - name: Setup tools
       uses: ./.github/actions/setup-tools
+      #{{- if not .Config.Shards }}#
       with:
         tools: pulumictl, pulumicli, ${{ matrix.language }}
+      #{{- end }}#
     - name: Prepare local workspace
       run: make prepare_local_workspace
     - name: Download bin
       uses: ./.github/actions/download-bin
+    #{{- if not .Config.Shards }}#
     - name: Download SDK
       uses: ./.github/actions/download-sdk
       with:
         language: ${{ matrix.language }}
     - name: Restore makefile progress
       run: make --touch provider schema build_${{ matrix.language }}
+    - name: Install dependencies
+      run: make install_${{ matrix.language}}_sdk
+    #{{- else }}#
+      #{{- range $_, $language := .Config.Languages }}#
+    - name: Download #{{ $language }}# SDK
+      uses: ./.github/actions/download-sdk
+      with:
+        language: #{{ $language }}#
+    - name: Restore makefile progress
+      run: make --touch provider schema build_#{{ $language }}#
+    - name: Install dependencies
+      run: make install_#{{ $language }}#_sdk
+      #{{- end }}#
```

I think we can make this less repetitive in the generated code by doing the touch and install outside the loop and leveraging the phony grouping targets.

```suggestion
      #{{- range $_, $language := .Config.Languages }}#
    - name: Download #{{ $language }}# SDK
      uses: ./.github/actions/download-sdk
      with:
        language: #{{ $language }}#
      #{{- end }}#
    - name: Restore makefile progress
      run: make --touch provider schema build_sdks
    - name: Install dependencies
      run: make install_sdks
```

---

Additionally, we could allow the "Download SDK" action to accept multiple languages (e.g. comma separated or something) to make the call sight cleaner and more understandable.

---

Totally forgot about `install_sdks`, good call!